### DISCUSSION TITLE
lean angle

### DISCUSSION THREAD

**Discussion by apropos** [time4tea/gopro-dashboard-overlay#80] [Q&A] [answered]
How do I go about creating a lean angle indicator widget? I found the following on an online forum. It seems lean angle/roll can be calculated from the 3 accelerometer readings.

accelerationX = (signed int)(((signed int)rawData_X) * 3.9);
accelerationY = (signed int)(((signed int)rawData_Y) * 3.9);
accelerationZ = (signed int)(((signed int)rawData_Z) * 3.9);
pitch = 180 * atan (accelerationX/sqrt(accelerationY*accelerationY + accelerationZ*accelerationZ))/M_PI;
roll = 180 * atan (accelerationY/sqrt(accelerationX*accelerationX + accelerationZ*accelerationZ))/M_PI;

**Comment by time4tea** [comment, marked as answer]
Hey thanks for this. I can probably add it. It would be interesting to think of how derived values could be added in a configurable way.

**Comment by time4tea** [comment]
So in 0.67.0 there is pitch, roll, and yaw... its definitely experimental! I'm not sure if it's "useful" yet, but you can make a `bar` of it or perhaps even a gauge. Let me know.

**Reply by apropos** [reply]
thanks, i will give it a try!

**Comment by time4tea** [comment]
@apropos - was this any use? If you have a test file that you'd be able to share it might help in the creation of a suitable widget... or any ideas about how such a widget should look? Thanks! if you are able - please use WeTransfer or similar to send to gopro-overlay [at] time4tea.net - but also, no worries if you don't feel able to share.

**Comment by Camses** [comment]
Good afternoon. Are there any additional steps to display the ori.pitch values? I've added the parameter to the xml, and when I enter the command to create a video, it's also listed in the field, but there are no values (a dash). However, when I use the same video in another program, the tilt displays correctly (GoPro 11). 

        <composite x="2000" y="0" name="ori.pitch">
            <component type="text" x="70" y="0" size="16">PITCH</component>
            <component type="metric" x="70" y="18" metric="ori.pitch" dp="2" size="32" />
        </composite>

**Comment by time4tea** [comment]
Hi - thanks for your interest in the program. 

Calculating the lean angle requires loading more of the datafile which is a bit slow - a GoPro has very high resolution information for camera angle and acceleration, as this is how its able to do image stabilisation. I seem to remember that adding:

--load CORI

to the command line will load the extra information. This code hasn't been well exercised, so please do report if it works. Thanks!

**Comment by Camses** [comment]
It works, thank you. Now I need to figure out how to convert this data into angles or degrees. The standard gradient parameter doesn't accurately display the slope, often leading to errors or significant delays. If the camera is mounted on a bicycle, it becomes convenient to analyze the data based on the camera's tilt and mark the inclines on the map.

**Comment by time4tea** [comment]
That should be a standard unit conversion.. sorry its probably a bit cryptic. Rotation in the app is measured in radians

https://github.com/time4tea/gopro-dashboard-overlay/tree/main/docs/xml/examples/04-metrics

so, to put the angle in degrees, you can use `unit="degrees"`  in the metric component.

there is currently no way to account for the mounting rotation of the camera though..
